src/gossip/data.zig

```diff
@@ -1084,26 +1087,30 @@ pub const ContactInfo = struct {
         try sanitizeWallclock(self.wallclock);
     }
 
-    pub fn getSocket(self: *const Self, key: u8) ?SocketAddr {
-        if (self.cache[key].eql(&SocketAddr.UNSPECIFIED)) {
+    pub fn getSocket(self: *const Self, key: socket_tag) ?SocketAddr {
+        const maybe_socket = &self.cache[@intFromEnum(key)];
+        if (maybe_socket.eql(&SocketAddr.UNSPECIFIED)) {
             return null;
         }
-        return self.cache[key];
+        return maybe_socket.*;
     }
 
-    pub fn setSocket(self: *Self, key: u8, socket_addr: SocketAddr) !void {
+    pub fn setSocket(self: *Self, key: socket_tag, socket_addr: SocketAddr) !void {
         self.removeSocket(key);
 
-        var offset = socket_addr.port();
-        var index: ?usize = null;
-        for (self.sockets.items, 0..) |socket, idx| {
-            offset = std.math.sub(u16, offset, socket.offset) catch {
-                index = idx;
-                break;
-            };
-        }
+        const offset: u16, const index: ?usize = blk: {
+            var offset = socket_addr.port();
+            const index = for (self.sockets.items, 0..) |socket, idx| {
+                offset = std.math.sub(u16, offset, socket.offset) catch break idx;
+            } else null;
+            break :blk .{ offset, index };
+        };
```

Comment:
imo i find the original loop approach more readable because its consistent with a lot of other languages and doesnt require a `blk:` - @dnut wdyt